### ISSUE TITLE
Basemap language: 1. Add French language maps; 2. Remove foreign language from English maps

### ISSUE THREAD

**Issue by jvanulde** [OpenDRR/riskprofiler#31]
French language base map required.

**Comment by phil-evans** [comment]
swapped to google's basemap which can be requested with language specified

**Comment by jvanulde** [comment]
@phil-evans are there any licensing issues we need to look after for the Google basemap?

**Comment by anthonyfok** [comment]
Curiously, Japanese names start appearing in "English" mode:

![image](https://user-images.githubusercontent.com/1274764/178374608-dec15075-7d31-4ccf-af3a-b86bbcc67708.png)

The Japanese place names seem to be limited to the areas surrounding Vancouver though, perhaps a Google feature to cater to higher number of Japanese tourists and/or residents in the Greater Vancouver area?

Tested with https://opendrr.github.io/riskprofiler-cms/risks/index.html#eAALt_Asset on my computer running Google Chrome in Debian GNU/Linux (sid), even in Incognito Mode and as Guest.

But I don't see the Japanese place names an NRCan laptop running Google Chrome on Windows 10 though.  Hmm...

**Comment by anthonyfok** [comment]
> @phil-evans are there any licensing issues we need to look after for the Google basemap?

We're _probably_ OK?  But we'll need to examine Google's guidelines more carefully, e.g. at https://about.google/brand-resource-center/products-and-services/geo-guidelines/#google-maps-web-and-apps

**Comment by wongstea** [comment]
We made a change to the Google basemap that _should_ fix the Japanese text issue. Everyone take a look at the site on your end and we'll touch base at the meeting tomorrow to see if anyone is still experiencing the issue.

**Comment by anthonyfok** [comment]
@wongstea and @phil-evans, thank you for looking into the issue!

With help from Leaflet documentation (searching for the keyword `tileLayer`), I think finally found how you (partially) fixed the basemap language in rp_scenarios.js and rp_risks.js

```diff
 	// BASEMAP
 
-	var basemap_URL = 'https://mt1.google.com/vt/lyrs=m&x={x}&y={y}&z={z}',
+	var basemap_URL = 'https://mt1.google.com/vt/lyrs=m&x={x}&y={y}&z={z}&region=CA',
 				basemap_att = 'Map data © 2022 Google | <a href="https://www.google.com/intl/en-CA_US/help/terms_maps/" target="_blank">Terms of use</a>'
 
 	if ($('body').hasClass('lang-fr')) {
```

Setting the region to CA (Canada) is a great start!  It does remove the Japanese text from the basemap on my English Windows 10 system.  However, I was able to get the basemap to display Chinese on RiskProfiler English pages when I set my system language to Chinese.  No such problem on RiskProfiler French pages, and that is because you specified `&hl` for French:

```js
	if ($('body').hasClass('lang-fr')) {
		basemap_URL += '&hl=fr'
		basemap_att = 'Map data © 2022 Google | <a href="https://www.google.com/intl/en-CA_US/help/terms_maps/" target="_blank">Terms of use</a>'
	}
```

without which (like currently for English pages), Google Maps would read from the [`Accept-Language` HTTP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language) and show place names accordingly, i.e. potentially different results for different users. That could be desirable for some web applications, but probably not for RiskProfiler, at least not right now.

I suggest changing the above `if` block to something like the following so that the language (`&hl` parameter) is always explicitly defined; and since Google does provide "French (Canadian)" language code `fr-CA`, let's use that too:

```js
	if ($('body').hasClass('lang-fr')) {
		basemap_URL += '&hl=fr-CA'
		basemap_att = 'Données cartographiques © 2022 Google | <a href="https://www.google.com/intl/fr_ca/help/terms_maps/" target="_blank">Conditions d'utilisation</a>'
	} else {
		basemap_URL += '&hl=en'
	}
```

Kudos to @wkhchow who found the addition of `&hl=en` fixed the problem for him when he tested in QGIS.
(Sorry, it took me a while to find the chat log.)  He also found this discussion on GIS Stack Exchange: https://gis.stackexchange.com/questions/323724/setting-google-street-basemap-language-in-qgis

See also https://developers-dot-devsite-v2-prod.appspot.com/maps/documentation/javascript/demos/localization which is a very nice demo for experimenting with map localization.

**Comment by anthonyfok** [comment]
@jvanulde shared this gem at last Thursday's meeting:

> National basemap services (WMS) in French and English:
> https://maps-cartes.services.geo.ca/server2_serveur2/rest/services/BaseMaps
> Note that the text and base layers are separate layers
> We can use the 3857 which is same at 900913

And it is definitely something worth looking into, not only because it is our very own Canadian national basemap, but also:

> We also need to load the placename labels (e.g. municipal) on top of the data.

And our national basemap services with separate text layer would be really helpful.

Let's briefly discuss at today's meeting on how and when we may want to incorporate this into a future version of RiskProfiler.
Cheers!

**Comment by anthonyfok** [comment]
Fixed in [v1.0-API_v1.4.0-20220825](https://github.com/OpenDRR/riskprofiler-cms/releases/tag/v1.0-API_v1.4.0-20220825) by @phil-evans in commits OpenDRR/h7-riskprofiler@9f01a1cc35145b7ab5d8938ee1151c649d7f393f and OpenDRR/h7-riskprofiler@69ba7d39e8f9386da81fbfe0a0f22239a576696c.
